src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/BitbucketServerAPIClient.java

```diff
@@ -203,53 +203,47 @@ public String getRepositoryName() {
     @Override
     public String getRepositoryUri(@NonNull BitbucketRepositoryType type,
                                    @NonNull BitbucketRepositoryProtocol protocol,
-                                   @CheckForNull Integer protocolPortOverride,
+                                   @CheckForNull String cloneLink,
                                    @NonNull String owner,
                                    @NonNull String repository) {
         switch (type) {
             case GIT:
-                URL url;
+                URI baseUri;
                 try {
-                    url = new URL(baseURL);
-                } catch (MalformedURLException e) {
-                    throw new IllegalStateException("Server URL is not a valid URL", e);
+                    baseUri = new URI(baseURL);
```

Comment:
I'm a bit reluctant to do this also for HTTP cloneLinks, since we have to support the fork case, where we need to replace the last two path segments. Since HTTP links may very well include sub-paths and also need the additional "/scm/" path segment in between, this would include a lot of reverse engineering of the original cloneLink to just replace the owner and repo part of it.

Moreover, I think you cannot configure Bitbucket Server in another way than to host HTTP git URLs at that exact `server-base-url/scm/owner/repo` URL, so I don't see any benefit in using cloneLinks there.

If you still think that we should do it, could you please give a hint on how you'd implement the replacement part for HTTP cloneLinks?

Reply:
I was more looking for the reduced logic to calculate something as a basic URL 🤷‍♂️  But I see your point 🤭